push_as_submodule.sh

```diff
@@ -0,0 +1,149 @@
+#!/bin/bash
+# MIT License
+#
+# Copyright (c) 2022 Red Hat, Inc.
+
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+#----------------------------------
+# parameters to configure by the user
+
+declare -a all_sclorg_images=(\
+"s2i-base"
+"mysql"
+"mariadb"
+"postgresql"
+"redis"
+"s2i-php"
+"s2i-python"
+"s2i-perl"
+"s2i-ruby"
+"varnish"
+"nginx"
+"s2i-nodejs"
+"httpd"
+)
+
+branch=master
+remote=origin
+commit_msg="Update common submodule to current latest $remote/$branch commit"
+
+#----------------------------------
+
+function exit_on_err() {
+  local ret=$1
+  local reason=${2:-"unknown"}
+  if [[ $ret -ne 0 ]]; then
+    echo "ERR: Exiting for reason: $reason" 2>&1 && exit 42
+  fi
+}
+
+function clone_repo() {
+  local image=$1
+  [[ -z $image ]] && exit_on_err 1 "image name for cloning repository is missing"
+  local container="${image}-container"
+  local url="git@github.com:sclorg/${container}.git"
+  git clone "$url" "$image"
+  exit_on_err $? "git clone failed"
+  cd "$image" || return 1
+  git submodule update --init
+  exit_on_err $? "submodule init failed"
+  cd ..
+}
+
+function update_submodule() {
+  local sha=$1
+  [[ -z "$sha" ]] && exit_on_err 1 "commit SHA is missing for updating submodule"
+  cd common/ || return 1
```

Comment:
the return 1 is not needed, because if `git submodule update --init` fails, then common directory does not exist. See `clone_repo` function.

Reply:
This is only a correctness check. I think it is valid. Maybe someone would decide to rename `common` and then we would need this check.

Could you please explain your though more, maybe I misunderstood?